gravitee-apim-rest-api/gravitee-apim-rest-api-management/gravitee-apim-rest-api-management-rest/src/main/java/io/gravitee/rest/api/management/rest/resource/ApiResource.java

```diff
@@ -482,6 +462,46 @@ public Response deprecated_updateApiWithDefinition(@RequestBody @Valid @NotNull
     }
 
     @PUT
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("import")
+    @Operation(
+        summary = "Update the API with an existing API definition in JSON format either with json or via an URL",
+        description = "User must have the MANAGE_API permission to use this service"
+    )
+    @ApiResponse(
+        responseCode = "200",
+        description = "API successfully updated from API definition",
+        content = @Content(mediaType = MediaType.APPLICATION_JSON, schema = @Schema(implementation = ApiEntity.class))
+    )
+    @ApiResponse(responseCode = "500", description = "Internal server error")
+    @Permissions({ @Permission(value = RolePermission.API_DEFINITION, acls = RolePermissionAction.UPDATE) })
+    public Response updateApiWithDefinition(@RequestBody @Valid @NotNull JsonNode apiDefinition) {
+        return updateApiWithDefinitionOrUrl(apiDefinition);
+    }
+
+    @PUT
+    @Consumes(MediaType.TEXT_PLAIN)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("import-url")
+    @Operation(
+        summary = "Update the API with an existing API definition in JSON format either with json or via an URL",
+        description = "User must have the MANAGE_API permission to use this service"
+    )
+    @ApiResponse(
+        responseCode = "200",
+        description = "API successfully updated from API definition",
+        content = @Content(mediaType = MediaType.APPLICATION_JSON, schema = @Schema(implementation = ApiEntity.class))
+    )
+    @ApiResponse(responseCode = "500", description = "Internal server error")
+    @Permissions({ @Permission(value = RolePermission.API_DEFINITION, acls = RolePermissionAction.UPDATE) })
+    public Response updateApiWithUrl(@RequestBody @Valid @NotNull String apiDefinitionOrUrl) {
+        return updateApiWithDefinitionOrUrl(apiDefinitionOrUrl);
+    }
+
+    @Deprecated(since = "3.18.0", forRemoval = true)
```

Comment:
Should we add an upgrade note about endpoint deprecation, in order to remove it as soon as possible ?